### PR TITLE
Improve performance of leak tracking.

### PR DESCRIPTION
**Pull request by polina-c**
Decrease number of full gc cycles. 

Prerequisite: https://github.com/flutter/flutter/pull/130951

### CODE REVIEW

File: packages/flutter/test/foundation/leak_tracking.dart [hunk 1]
```diff
@@ -122,6 +122,7 @@ Future<void> _withFlutterLeakTracking(
         asyncCodeRunner: asyncCodeRunner,
         leakDiagnosticConfig: config.leakDiagnosticConfig,
         shouldThrowOnLeaks: false,
+        gcCountBuffer: 1,
```
Comment (goderbauer) [inline packages/flutter/test/foundation/leak_tracking.dart:125]: Are there any downsides to this?

Comment (goderbauer) [reply packages/flutter/test/foundation/leak_tracking.dart:125]: What does this property actually control? It's documentation wasn't all that helpful: https://github.com/dart-lang/leak_tracker/blob/6e3f57c8ce7609882ea2209b3d07937b73823e23/pkgs/leak_tracker/lib/src/leak_tracking/orchestration.dart#L48

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/test/foundation/leak_tracking.dart (hunk 1 above)
Comment (polina-c) [reply packages/flutter/test/foundation/leak_tracking.dart:125]: Improvement for documentation: https://github.com/dart-lang/leak_tracker/pull/100

Comment (polina-c) [reply packages/flutter/test/foundation/leak_tracking.dart]: Downside can be false positive. Already got some failures, so increased to 2.

**Pull request closed without merging**